Use eth2-clients / slashing-protection-interchange-tests & add support for v5 slashing protection interchange

3xtr4t3rr3str14l opened the pull request:
resolves #2001 and adds support for v5 interchange (as spec'd here: https://eips.ethereum.org/EIPS/eip-3076)

dapplion commented on packages/validator/test/e2e/slashing-protection-interchange-tests/tests/tests/generated/duplicate_pubkey_not_slashable.json:
```diff
@@ -0,0 +1,88 @@
+{
```
I don't think you want to commit this files to the source code, but download them on demand when testing

3xtr4t3rr3str14l replied:
so should we maybe delete them when the tests are done running?  or just add them to the .gitignore?

wemeetagain reviewed (dismissed):
lgtm

dapplion commented on packages/validator/test/e2e/slashing-protection-interchange-tests/run.test.ts:
```diff
@@ -12,92 +12,98 @@ import {
   InvalidBlockError,
 } from "../../../src/slashingProtection";
-import {downloadTests, loadTestCases as newLoadTestCases} from "./downloadTest";
+import {downloadTests, loadTestCases} from "./index.test";
```
Importing from .test files in not very standard. Consider refactoring

3xtr4t3rr3str14l replied:
it didn't really need to be named a test file, so i just renamed it.

dapplion committed:
convert slashing interchange protection tests to local tests (no NPM package)

convert slashing interchange protection tests to local tests (no NPM package)

remove console.log

add v5 interchange support and tests

.gitignore downloaded spi tests

move downloadTest

ensure we are downloading spi tests before running

rm @chainsafe/slashing-protection-interchange-tests

update yarn.lock

rename index.test.ts to downloadTests.ts

package.json rm old eventsource types

dapplion committed:
Review PR
